From fe7d0fa216c1ebea42e3391b18e9192eca5fa20f Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Mon, 22 Mar 2021 09:27:16 +0000
Subject: [PATCH] ARM: dts: bcm2711: Add aliases for additional SPIs

Without aliases for the new SPI interfaces in BCM2711, spidev instances
will be allocated sequential numbers that may not match the number of
the physical interface.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/bcm2711-rpi-4-b.dts | 4 ++++
 arch/arm/boot/dts/bcm2711-rpi-400.dts | 4 ++++
 arch/arm/boot/dts/bcm2711-rpi-cm4.dts | 4 ++++
 3 files changed, 12 insertions(+)

--- a/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-4-b.dts
@@ -355,6 +355,10 @@
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		spi3 = &spi3;
+		spi4 = &spi4;
+		spi5 = &spi5;
+		spi6 = &spi6;
 		/delete-property/ intc;
 	};
 
--- a/arch/arm/boot/dts/bcm2711-rpi-400.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-400.dts
@@ -336,6 +336,10 @@
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		spi3 = &spi3;
+		spi4 = &spi4;
+		spi5 = &spi5;
+		spi6 = &spi6;
 		/delete-property/ intc;
 	};
 
--- a/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
+++ b/arch/arm/boot/dts/bcm2711-rpi-cm4.dts
@@ -323,6 +323,10 @@
 		i2c4 = &i2c4;
 		i2c5 = &i2c5;
 		i2c6 = &i2c6;
+		spi3 = &spi3;
+		spi4 = &spi4;
+		spi5 = &spi5;
+		spi6 = &spi6;
 		/delete-property/ intc;
 	};
 
